targets/posix/Include/nanoPAL.h

```
@@ -0,0 +1,61 @@
+        return 1;
+    size_t n = count < size - 1 ? count : size - 1;
+    memcpy(dst, src, n);
+    dst[n] = '\0';
+    // return 1 if the source was longer than the buffer allows (truncation occurred)
+    return (count >= size) ? 1 : 0;
+}
+
+// __isnanf - POSIX equivalent (macOS/Linux use isnan() for float too)
+#include <cmath>
+#ifndef __isnanf
+#define __isnanf(x) std::isnan(static_cast<float>(x))
```

Clang (14.0.6)

[warning] declaration uses identifier '__isnanf', which is a reserved identifier

(bugprone-reserved-identifier)